aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -44,11 +44,15 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
```

Comment:
This was the same conclusion that @rjuliano and I reached. I think we both actually like the Stream APIs a little better, but since we already have RxJava here, you can do very similar stuff:
```java
Observable.fromArray(one, two, three)
    .blockingForEach(item -> {
    });
```
Or:
```java
Observable.fromArray(one, two, three)
    .map(item -> {}) // This is like `.map(..)` in Stream APIs
    .toList() // This is like the `.collect(..)` in Stream APIs
    .blockingGet();
```

Reply:
Got it, will make the changes